### PR TITLE
add helper method to serialize the filter to byte vector. 

### PR DESCRIPTION
**Pull request by poonai**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -402,71 +403,64 @@ impl Xor8 {
...
-            .zip(&Xor8::SIGNATURE_V1)
-            .any(|(a, b)| a != b)
-        {
+        let mut data = Vec::new();
+        f.read_to_end(&mut data)?;
+        Self::from_bytes(data)
+    }
+
+    pub fn to_bytes(&self) -> Vec<u8> {
+        let required_capcity = Self::METADATA_LENGTH + self.finger_prints.len();
```
Comment (prataprc) [inline src/lib.rs]: we can rename `required_capcity` to just `capacity`.

File: src/lib.rs [hunk 2]
```diff
@@ -402,71 +403,64 @@ impl Xor8 {
...
+        let mut buf: Vec<u8> = Vec::with_capacity(required_capcity);
+        unsafe {
+            buf.set_len(required_capcity);
```
Comment (prataprc) [inline src/lib.rs]: We can avoid this.

File: src/lib.rs [hunk 3]
```diff
@@ -402,71 +403,64 @@ impl Xor8 {
...
+        }
+        buf[..4].copy_from_slice(&Xor8::SIGNATURE_V1);
+        buf[4..12].copy_from_slice(&self.seed.to_be_bytes());
```
Comment (prataprc) [inline src/lib.rs]: we can replace copy_from_slice with extend_from_slice().

**Review by prataprc — CHANGES_REQUESTED** [review]
I have some specific comments and also there is one general comment.

a. Do you think it make sense to have to_bytes() return with signature_v1 in the header. I might be okay with it. But in general serialization is kept separate from version-prefixing. Just a thought.

File: src/lib.rs (hunk 1 above)
Comment (poonai) [reply src/lib.rs]: done

**Comment by poonai** [comment]
Some folks probably want to store the bytes in kv. Having version number in bytes assert the compactibility. If you feel that is off, I'm happy to remove that.

File: src/lib.rs (hunk 3 above)
Comment (poonai) [reply src/lib.rs]: extend_from_slice is slower. Because it's iterate over the incoming slice and pushes one by one. When in copy_from_slice use memcpy.

Comment (prataprc) [reply src/lib.rs]: Please check this link https://doc.rust-lang.org/stable/src/alloc/vec.rs.html#1259

both extend_from_slice() and copy_from_slice() use copy_nonoverlapping intrinsic. Let me know if I am mistaken.

Comment (poonai) [reply src/lib.rs]: Done

**Pull request merged**